oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolver.java

```diff
@@ -126,4 +130,17 @@ private boolean isParameterTokenSupportedForRequest(HttpServletRequest request)
 		return ((this.allowFormEncodedBodyParameter && "POST".equals(request.getMethod()))
 				|| (this.allowUriQueryParameter && "GET".equals(request.getMethod())));
 	}
+
+	private String validateQueryParamter(String parameterToken) {
```

Comment:
I think it would help if the two methods `resolveFromAuthorizationHeader` and `resolveFromRequestParameters` looked like each other. What if you took the contents of this method and inlined them into `resolveFromRequestParameters` so that the two methods structurally read the same?

Possibly another way would be to change both methods to call a common validation method.

Reply:
I've updated the PR. It now uses a common validation method now.